internal/pkg/config/v1/config.go

```diff
@@ -314,7 +388,7 @@ func (c *Config) Validate() error {
 		if _, ok := c.ContextStates[context.Name]; !ok {
 			c.ContextStates[context.Name] = new(ContextState)
 		}
-		if *c.ContextStates[context.Name] != *context.State {
+		if !c.IsTest && !reflect.DeepEqual(*c.ContextStates[context.Name], *context.State) {
```

Comment:
unit tests can't pass this test as it's impossible to preset context in golden files. there'll be randomly generated salt and nonce in the context